### PR TITLE
fix: add number_sms_notifications translation

### LINKED ISSUE #11718: [CAL-2584] Missing translation for number_sms_notifications
### Issue Summary

Translation is missing, see here: 
Screenshot 2023-10-06 at 09 28 17

This was replaced with `number_text_notifications` when we introduced Whatsapp workflow actions, but workflows that existed before still have the default label `number_sms_notifications`. 

The text should be: Phone number (SMS notifications)

<sub>[CAL-2584](https://linear.app/calcom/issue/CAL-2584/missing-translation-for-number-sms-notifications)</sub>

### PR DESCRIPTION
**Pull request by Nikeshkumar-tk**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/11718

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/public/static/locales/en/common.json [hunk 1]
```diff
@@ -1534,7 +1534,7 @@
   "url_taken": "This URL is already taken",
   "problem_registering_domain": "There was a problem with registering the subdomain, please try again or contact an administrator",
   "team_publish": "Publish team",
-  "number_text_notifications": "Phone number (Text notifications)",
+  "number_text_notifications": "Phone number (SMS notifications)",
```
Comment (CarinaWolli) [inline apps/web/public/static/locales/en/common.json:1537]: That doesn't fix it. We need a new translation key `number_sms_notifications`
We want to have both

**Review by CarinaWolli — DISMISSED** [review]

File: apps/web/public/static/locales/en/common.json (hunk 1 above)
Comment (Nikeshkumar-tk) [reply apps/web/public/static/locales/en/common.json]: Okay.I will update it

Comment (Nikeshkumar-tk) [reply apps/web/public/static/locales/en/common.json]: I have updated the PR with new changes.

File: apps/web/public/static/locales/en/common.json [hunk 2]
```diff
@@ -1535,6 +1535,7 @@
   "problem_registering_domain": "There was a problem with registering the subdomain, please try again or contact an administrator",
   "team_publish": "Publish team",
   "number_text_notifications": "Phone number (Text notifications)",
+  "number_sms_notifications":"Phone number (SMS notifications)",
```
Comment (PeerRich) [inline apps/web/public/static/locales/en/common.json:1538]:
```suggestion
  "number_sms_notifications": "Phone number (SMS notifications)",
```

**Review by PeerRich — APPROVED** [review]

**Pull request merged**